plugins: add fullscreen plugin

sanga opened the pull request:
I suppose at this point this is more of a request of interest (or lack of interest).

Tested manually and works. However should have a test added. Also it's somewhat unuseful at the moment (at least for me) as it won't work in an iframe which means that currently (I've filed a bug with the jupyter guys) it won't work in jupyter.

But, basically. Should I bother polishing this a bit more (adding a test etc)?

ocefpaf commented:
@sanga this is definitely worth polishing! Nice addition.

ocefpaf commented on folium/plugins/fullscreen.py:
```diff
@@ -0,0 +1,54 @@
+
+Note: does not work from within an iframe, and because of this, does not
+work in jupyter notebook
+"""
```
I would put this in the class docstring. Add here just a quick description like `Adds fullscreen button to your maps.`

ocefpaf commented on folium/plugins/fullscreen.py:
```diff
@@ -0,0 +1,54 @@
+
+        figure.header.add_children(
+            JavascriptLink(
+                "https://cdnjs.cloudflare.com/ajax/libs/"
```
It is fine to use long lines for URLs (and more readable/copy-n-pastable :wink:). Just add a `  # noqa` comment at the end to tell the linter to skip it.

sanga commented:
Addressed all comments (I think). Did a little cleanup. Added (a rather soft) test. Added an example. And did a couple of 'drive by' fixes elsewhere. Let me know if you'd rather have the fixes in a separate branch. Oh and I force pushed with clean git history (let me know if you'd rather I didn't)

sanga commented:
And force pushed a few cleanups to the fullscreen test

ocefpaf commented on folium/plugins/fullscreen.py:
```diff
@@ -0,0 +1,54 @@
+    Adds a fullscreen button to your map.
+
+    Note: does not work from within an iframe, and because of this, does not
+    work in jupyter notebook
```
You mentioned you filed a bug about this, right? Can you add a cross-reference link here so people can follow the status of that?

ocefpaf commented on tests/plugins/test_fullscreen.py:
```diff
@@ -0,0 +1,19 @@
+    out = m._parent.render()
+
+    # verify that the fullscreen control was rendered
+    assert 'L.control.fullscreen().addTo' in out
```
:+1:

ocefpaf commented:
LGTM. Only 1 minor comment.

@BibMartin if you find some time to take a quick look and, this is OK for you, go ahead and merge. Otherwise I will merge this next week once I get back in office.

ocefpaf commented:
@sanga do you mind also adding an entry in the changelog file?

sanga commented:
So I figured out who was rendering that iframe element (to which the allowfullscreen attr needed to be added). It wasn't jupyter actually. It's branca:

https://github.com/python-visualization/branca/pull/13

I've just noticed that I missed your comment about the changelog. Change coming shortly

ocefpaf commented:
> So I figured out who was rendering that iframe element (to which the allowfullscreen attr needed to be added). It wasn't jupyter actually. It's branca:

Cool. Since that is an in-house issue let's get python-visualization/branca#13 sorted out first before we merge this. I am in a conference and I will take a look at this as soon as I get back. But it looks all good already. Thanks @sanga!

sanga commented:
python-visualization/branca#13  is merged so this should be mergable now.

ocefpaf commented:
Just used this plugin for a GIS lecture using `Jupyter` notebooks and `folium`. Thanks @sanga!

sanga commented:
no worries! :)
